packages/website/docs/using-react.md

```diff
@@ -0,0 +1,179 @@
+import { render } from 'react-dom';
+
+export function Autocomplete(props) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return undefined;
+    }
+  }, [props]);
+
+  return (
+    <div
+      ref={containerRef}
+    />
+  );
+}
+```
+
+### Mounting the autocomplete
+
+Now that you have access to the DOM through the `containerRef` object, you can create and mount the Autocomplete instance. Upon instantiation, you can include any desired [Autocomplete options](basic-options) and rely on `props` to pass any options you want to remain configurable.
+
+The example component below sets only the [`container`](autocomplete-js/#container) option. It specifies where to mount your Autocomplete component, but lets all other [options](basic-options) get configured through props.
+
+**You must also pass the `renderer` and `render` parameters.** This is because the default Autocomplete implementation uses [Preact's](https://preactjs.com/) version of `createElement`, `Fragment` and `render`. Without providing React's version of these, the Autocomplete instance won't render the views properly.
+
+Don't forget to [clean up the effect](https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect) by returning a function that destroys the Autocomplete instance.
+
+```jsx title="Autocomplete.jsx"
+import { autocomplete } from '@algolia/autocomplete-js';
+import React, { createElement, Fragment, useEffect, useRef } from 'react';
+import { render } from 'react-dom';
+
+export function Autocomplete(props) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return undefined;
+    }
+
+    const search = autocomplete({
+      container: containerRef.current,
+      renderer: { createElement, Fragment },
+      render({ children }, root) {
+        render(children, root);
+      },
+      ...props,
+    });
+
+    return () => {
+      search.destroy();
+    };
+  }, [props]);
+
+  return (
+    <div
+      ref={containerRef}
+    />
+  );
+}
+```
+
+## Using the component
+
+Now that you've created an `<Autocomplete/>` component, you can use it in your React application.
+
+As previously state, the example component sets only the [`container`](autocomplete-js/#container) option, but lets all other [options](basic-options) get configured through props.
+
+The usage below sets [`openOnFocus`](autocomplete-js#openonfocus) and [sources](sources) through props. This example uses an [Algolia index](https://www.algolia.com/doc/faq/basics/what-is-an-index/) as a [source](sources), but you could use anything else you want, including [plugins](plugins). For more information on using Algolia as a source, check out the [Getting Started guide](getting-started).
+
+
+```jsx title=App.jsx"
+import React, { createElement } from 'react';
+import { Autocomplete } from './components/Autocomplete';
+import { ProductItem } from './components/ProductItem';
```

Comment:
Can we move these two local imports after the package imports?

Reply:
https://github.com/algolia/autocomplete.js/pull/442/commits/78bd3c2913527485b9c91dec424011eb7ecee7d8